File: api/src/main/java/com/prep/api/model/interesse/Interesse.java
package com.prep.api.model.interesse;

import com.prep.api.model.ModelAbstract;
import com.prep.api.model.funcionario.Funcionario;
import com.prep.api.model.modulo.Modulo;
import com.prep.api.model.modulo.ModuloJPA;
import com.prep.api.model.pessoa.Pessoa;

import java.util.Date;
import java.util.List;
import java.util.Set;


public class Interesse  extends ModelAbstract {

    protected Long id;
    protected String status, tipo;
    protected Date dataDoCadastro;
    protected Pessoa pessoa;
    protected Funcionario funcionario;
    protected Set<? extends Modulo> modulos;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public Date getDataDoCadastro() {
        return dataDoCadastro;
    }

    public void setDataDoCadastro(Date dataDoCadastro) {
        this.dataDoCadastro = dataDoCadastro;
    }

    public Pessoa getPessoa() {
        return pessoa;
    }

    public void setPessoa(Pessoa pessoa) {
        this.pessoa = pessoa;
    }

    public Funcionario getFuncionario() {
        return funcionario;
    }

    public void setFuncionario(Funcionario funcionario) {
        this.funcionario = funcionario;
    }
}

File: ui/src/app/modules/interesse/models/interesse.ts
import { Funcionario } from '../../funcionario/models/funcionarios';
import { Modulo } from '../../curso/models/modulo';
import { Pessoa } from '../../pessoa/models/pessoa';
import { InteresseStatus } from './interesse.status';

export class Interesse {

  public id: number;
  public status: string;
  public tipo: string;
  public dataDoCadastro: Date;
  public pessoa: Pessoa;
  public funcionario: Funcionario;
  public modulos: Array<Modulo>;
  // TODO: Arrumar
  // public modulos: Array<Modulo>;

  public static newInteresse(): Interesse {
    const i = new Interesse();
    i.status = InteresseStatus.PENDENTE;
    i.dataDoCadastro = new Date();
    i.modulos = new Array<Modulo>();
    return i;
  }
}
